### PR TITLE
Allow Checkout to work with sudo setup.sh

### PR DESCRIPTION
**Pull request by pboyd04**

### CODE REVIEW

File: bin/repos [hunk 1]
```diff
@@ -26,14 +26,14 @@ else
...
+git_clone_or_update "https://github.com/$githubuser/CommonCode.git" src/common
 action "init / update common code submodules"
 ( cd src/common && git submodule init && git submodule update  )
 ok
 
-git_clone_or_update "git@github.com:$githubuser/Profiles.git" src/profiles
+git_clone_or_update "https://github.com/$githubuser/Profiles.git" src/profiles
 
-git_clone_or_update "git@github.com:$githubuser/SecureFramework.git" src/secure
+git_clone_or_update "https://@github.com/$githubuser/SecureFramework.git" src/secure
```
Comment (russelltsherman) [inline bin/repos]: @ symbol unnecessary in this context

**Comment by russelltsherman** [comment]
please explain the advantage you see of using http protocol over git ssh..

**Comment by pboyd04** [comment]
As I stated over in the CentOS issue here: https://github.com/BurningFlipside/DevSandbox/issues/88#issuecomment-647034271

On CentOS you have to run setup.sh as root to get the docker commands to work. When you run with sudo it mucks this checkout up if you use SSH since it tries to authenticate with the SSH keys of root instead of your user's SSH keys. You therefore end up with the repo clones failing silently. Switching to https fixes that issue.

File: bin/repos (hunk 1 above)
Comment (pboyd04) [reply bin/repos]: Fixed

**Comment by russelltsherman** [comment]
adding your user to the docker group should allow you to execute docker commands without sudo and remove the need for these changes..

https://docs.docker.com/engine/install/centos/

i have always used git ssh for authenticated repos and https for read only..
it is supposedly possible to use https for authenticated repos but i am unfamiliar with the process.. are you familiar with that process?

**Comment by pboyd04** [comment]
Yeah I tried that and it's not working on Fedora 30. Not sure why not and rather than try and debug that using sudo worked pretty well.

As far as the workflow is concerned it's almost identical except that everytime you push with https you have to provide your username and password. So one extra step on push.

**Comment by russelltsherman** [comment]
that is curious.. adding a user to the docker group has worked in every flavor of linux i have used. perhaps updating or reinstalling docker will correct that for you

I am not in favor of changing this repo code to work around one system which is not properly responding to a well documented solution..

i will set aside time to spin up a new vm using fedora 30 and see if i can reproduce the issue..

**Comment by pboyd04** [comment]
There are also environments where SSH doesn't work. For example, behind my work's firewall I have to clone with https because of the proxy. How about if I make this an optional switch instead of always? That way I can use the flow I need to and you can use the flow you like and it doesn't really impact anything?

**Comment by russelltsherman** [comment]
sure.. i dont want to force my habits and preferences on to other devs.. so sure.. git ssh vs https should be user selectable..

**Comment by russelltsherman** [comment]
i have spun up a new virtual machine using fedora 30

i have installed docker and observed that adding my user to the docker group removes the need to use sudo for docker commands as documented in docker setup instructions.

the addition of the group to my user did not take affect until i logged out and restarted my session

i will continue to investigate other setup issues

**Comment by pboyd04** [comment]
I did log out as instructed but haven't restarted my system in a year or so. I will do so when my girlfriend is next not on my plex...

**Comment by pboyd04** [comment]
Ah now I remember the docker group stuff doesn't exist with the default install of docker that cockpit and some other system internals depend on. You need the community edition of docker to do all that....

**Pull request closed without merging**